scripts/typeGeneration/convert.js

```diff
@@ -0,0 +1,135 @@
+      }
+    }),
+    {}
+  );
+};
+
+const sanitizeSchemas = (schemas, existingSchemas, service, counter = {}) =>
+  Object.entries(schemas).reduce(
+    (schemaAccu, [key, schema]) => {
+      let sanitizedKey = key;
+      if (existingSchemas[key]) {
+        schemaAccu.counter[key] = (schemaAccu.counter[key] ?? 1) + 1;
+        console.log(`duplicate schema, candidate for merge: ${key} (${service}) - total: ${schemaAccu.counter[key]}`);
+        sanitizedKey = `${key}${service[0].toUpperCase()}${service.substring(1)}`;
+        schemaAccu.renamedSchemas.push({ key, sanitizedKey });
+      }
+      schemaAccu.serviceSchemas[sanitizedKey] = schema;
+      return schemaAccu;
+    },
+    { renamedSchemas: [], serviceSchemas: {}, counter }
+  );
+
+const defaultManagementUrl = 'https://hosted.mender.io/api/management';
+const baseSpec = {
+  openapi: '3.0.3',
+  info: {
+    contact: { name: 'contact@mender.io' },
+    description: 'Combined API specification for the management features of the different Mender backend services, suitable for code generation applications',
+    title: 'Mender Management API',
+    version: '1.0'
+  },
+  tags: [{ name: 'Management API', description: 'used for management APIs' }],
+  servers: [{ url: defaultManagementUrl }],
+  paths: {},
+  components: { securitySchemes: {}, schemas: {} }
+};
+
+const processFiles = async root => {
+  const files = await getFiles(root);
+  const fileContents = await Promise.all(files.map(getFileContents));
+  return fileContents.reduce((accu, { fileData, service, versionSuffix }) => {
+    const { renamedSchemas, serviceSchemas, counter } = sanitizeSchemas(fileData.components.schemas, accu.components.schemas, service, accu.counter);
+    const basePath = fileData.servers[0].url.substring(defaultManagementUrl.length);
+    const scopedPaths = Object.entries(fileData.paths).reduce((pathAccu, [path, content]) => {
+      const sanitizedContent = sanitizeSchemaRefs(content, renamedSchemas);
+      pathAccu[`${basePath}${path}`] = repairPathSpec(sanitizedContent, service, versionSuffix);
+      return pathAccu;
+    }, {});
+    return {
+      ...accu,
+      components: {
+        ...accu.components,
+        ...fileData.components,
+        responses: { ...accu.components.responses, ...fileData.components.responses },
+        schemas: { ...accu.components.schemas, ...serviceSchemas },
+        securitySchemes: { ...accu.components.securitySchemes, ...fileData.components.securitySchemes }
+      },
+      counter,
+      paths: { ...accu.paths, ...scopedPaths }
+    };
+  }, baseSpec);
+};
+
+const generateTypeIndex = async () => {
+  const fileContent = await Deno.readTextFile('./generated/index.ts');
+  const lines = fileContent.split('\n').reduce((accu, line) => {
+    if (line.startsWith('/* ')) {
+      accu.push(line);
+    } else if (line.includes(`from './models/`)) {
+      accu.push(line.replace('/models', ''));
+    }
```

Comment:
I am sorry, I either need some comments here, or some explanation, what is the reason behind this?

Reply:
whoops, yes that is indeed confusing without context 🙈 - I'll adjust!